snapshots/overlay/overlay.go

```diff
@@ -292,7 +327,19 @@ func (o *snapshotter) Walk(ctx context.Context, fn snapshots.WalkFunc, fs ...str
 		return err
 	}
 	defer t.Rollback()
-	return storage.WalkInfo(ctx, fn, fs...)
+	return storage.WalkInfo(ctx, func(ctx context.Context, info snapshots.Info) error {
```

Comment:
Instead of always wrapping, just conditionally wrap to avoid checking on each call

Reply:
updated to conditionally wrap the function.